services/wfm/src/main/java/org/openkilda/wfm/topology/event/OfeLinkBolt.java

```diff
@@ -432,39 +433,27 @@ private void reportInvalidEvent(InfoData event) {
                 event.getClass().getName());
     }
 
-    private void handleSwitchEvent(Tuple tuple, SwitchInfoData switchData) {
+    private void handleSwitchEvent(Tuple tuple, InfoMessage infoMessage) {
+        SwitchInfoData switchData = (SwitchInfoData) infoMessage.getData();
         SwitchId switchId = switchData.getSwitchId();
-        SwitchChangeType state = switchData.getState();
-        logger.info("DISCO: Switch Event: switch={} state={}", switchId, state);
+        SwitchChangeType switchState = switchData.getState();
+        logger.info("DISCO: Switch Event: switch={} state={}", switchId, switchState);
 
-        if (state == SwitchChangeType.DEACTIVATED) {
-            passToTopologyEngine(tuple);
-        } else if (state == SwitchChangeType.ACTIVATED) {
+        if (switchState == SwitchChangeType.DEACTIVATED) {
+            passToNetworkTopologyBolt(tuple, infoMessage);
+        } else if (switchState == SwitchChangeType.ACTIVATED) {
             // It's possible that we get duplicated switch up events .. particulary if
             // FL goes down and then comes back up; it'll rebuild its switch / port information.
             // NB: need to account for this, and send along to TE to be conservative.
             discovery.handleSwitchUp(switchId);
         } else {
             // TODO: Should this be a warning? Evaluate whether any other state needs to be handled
-            logger.warn("SWITCH Event: ignoring state: {}", state);
+            logger.warn("SWITCH Event: ignoring state: {}", switchState);
         }
     }
 
-    /**
-     * Pass the original message along, to the Topology Engine topic.
-     */
-    private void passToTopologyEngine(Tuple tuple) {
-        String json = tuple.getString(0);
-        collector.emit(TOPO_ENG_STREAM, tuple, new Values(PAYLOAD, json));
-    }
-
-    private void passToTopologyEngine(Tuple tuple, InfoMessage message) {
-        try {
-            String json = Utils.MAPPER.writeValueAsString(message);
-            collector.emit(TOPO_ENG_STREAM, tuple, new Values(PAYLOAD, json));
-        } catch (JsonProcessingException e) {
-            logger.error("Error during json processing", e);
-        }
+    private void passToNetworkTopologyBolt(Tuple tuple, InfoMessage message) {
+        collector.emit(NETWORK_TOPOLOGY_CHANGE_STREAM, tuple, new Values(message));
```

Comment:
Can we use more high level object for interstorm communications? Do we really need message and message data wrappers here?

(minor)